conda/gateways/disk/permissions.py

```diff
@@ -64,3 +65,9 @@ def make_executable(path):
         chmod(path, S_IMODE(mode) | S_IXUSR | S_IXGRP | S_IXOTH)
     else:
         log.error("Cannot make path '%s' executable", path)
+
+
+def is_executable(path):
+    if isfile(path):  # for now, leave out `and not islink(path)`
+        return path.endswith(('.exe', '.bat')) if on_win else access(path, X_OK)
+    return False
```

Comment:
@mingwandroid Also slight modification to this function.  It interacts with `_do_softlink()`.  One thing I've learned for sure being bitten too many times by it... `os.access()` just doesn't work on Windows.